llm/gguf.go

```diff
@@ -327,6 +343,10 @@ func readGGUFV1Array(llm *gguf, r io.Reader) (a []any, err error) {
 		return nil, err
 	}
 
+	a := &array{
```

Comment:
Why not make this:

a := make([]any, 0, min(maxArraySize, int(n)))
...
a= append(a, e)
...
return a, nil
```

Note if we're returning a struct here we need to update `routes.go` because this affects the output format of `/api/show`

Before (correct):
```
"tokenizer.ggml.tokens": []
```

With this change:
```
"tokenizer.ggml.tokens": {}
```

Reply:
Ah! Good catch. I'll change the return. It would be nice to not use the empty interface all over. No need. The compiler can prevent this stuff. :)

Reply:
Under a little load, relaxing the maxArraySize to 10e9 changes the p99 from 0.6 to 0.9 and takes measurable pressure off GC.

Reply:
```

Reply:
What is the `array` type provide that a regular slice doesn't?

Reply:
Marking this as unresolved in case above comment is missed

Reply:
or perhaps I missed that point and it's having strong typing the return value, but isn't `[]any` equivalent in that case

Reply:
Ah this makes sense now – `size` is meant to be the actual size of the array, despite how much we are willing to read – I'll mark this as resolved again.